stdlib/public/Cxx/CxxOptional.swift

```diff
@@ -0,0 +1,43 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+public protocol CxxOptional<Wrapped> {
+  associatedtype Wrapped
+
+  func __convertToBool() -> Bool
+
+  var pointee: Wrapped { get }
+}
+
+extension CxxOptional {
+  public var hasValue: Bool {
```

Comment:
This property provides the only way to check if the `std::optional` has a value from Swift.